custom/enikshay/tasks.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+from collections import defaultdict
+from django.utils.dateparse import parse_datetime, parse_date
+
+from .case_utils import get_adherence_cases_between_dates_from_episode
+from corehq.apps.fixtures.models import FixtureDataItem
+
+
+DOMAIN = 'enikshay'
+DOSE_TAKEN_INDICATORS = [
+    'directly_observed_dose',
+    'unobserved_dose',
+    'self_administered_dos',
+]
+DAILY_SCHEDULE_ID = 'schedule_daily'
+CASE_TYPE_PERSON = 'person'
+
+
+def get_doses_data():
+    # return 'doses_per_week' by 'schedule_id' from the Fixture data
+    fixtures = FixtureDataItem.get_indexed_items(DOMAIN, DAILY_SCHEDULE_ID, 'schedule_id')
+    return dict((k, int(fixture['doses_per_week'])) for k, fixture in fixtures.items())
+
+
+def get_open_episode_cases():
+    """
+    get list of all open 'episode' type cases
+    """
+    case_accessor = CaseAccessors(DOMAIN)
+    case_ids = get_open_case_ids_in_domain_by_type(CASE_TYPE_PERSON)
+    return case_accessor.iter_cases(case_ids)
+
+
+def get_latest_adherence_case_for_episode(episode):
+    """
+    return open case of type 'adherence' reversed-indexed to episode and
+        with latest 'adherence_date' property
+    """
+    case_accessor = CaseAccessors(DOMAIN)
+    indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    latest_date = 0
+    latest_case = None
+    for case in indexed_cases:
+        adherence_date = parse_datetime(case.dynamic_case_properties().get('adherence_date'))
+        if (not case.closed and
+           case.type == CASE_TYPE_ADHERENCE and
+           adherence_date > latest_date):
+            latest_date = adherence_date
+            latest_case = adherence_date
+    return latest_case
+
+
+def index_by_adherence_date(adherence_cases):
+    """
+    inde
+    """
+    by_date = defaultdict(list)
+    for case in adherence_cases:
+        adherance_date = parse_date(case.dynamic_case_properties().get('adherence_date'))
+        by_date[adherance_date].append(case)
+    return by_date
+
+def get_adherence_schedule_date_start(episode_case):
+    return parse_datetime(case.dynamic_case_properties().get('adherence_schedule_date_start'))
+
+
+def update_adherence_properties():
+    # edge cases around datetime
+    PURGE_DATE = datetime.datetime.today() - datetime.timedelta(days=60)
+
+    for episode in get_open_episode_cases():
+        adherence_schedule_date_start = get_adherence_schedule_date_start(episode)
+        # Todo: What if adherence_schedule_date_start is None
```

Comment:
I don't think these calculations should be run in that case because I think that would indicated that an adherence schedule hasn't been selected. @ctsims to confirm.